### DISCUSSION TITLE
How to allow to update '^' and '~' versions only

### DISCUSSION THREAD

**Discussion by andreibuntsev** [renovatebot/renovate#9848] [Request Help]
Hi Gurus,

Could you please advise the best way to allow renovation for '^' and '~' versions only..
i.e. suppress update for 
    "memoize-one": "5.1.1",
    "moment": "2.24.0",
because they are specified as exact versions
but allow update for 
    "classnames": "~2.2.6",
    "file-saver": "^2.0.2",

I believe, I should use  "packageRules"."matchCurrentVersion"  option ?
Or to play with this one?
"packageRules"."updateTypes": ["minor", "patch", "pin", "digest"]  ?

Thank you

**Comment by rarkins** [comment]
You could just disable those dependencies using `ignoreDeps`

**Reply by andreibuntsev** [reply]
Should I specify every package name I need to exclude?  What if I have 200 packages there? 
I would like to create a rule based on the version value

**Reply by viceice** [reply]
https://docs.renovatebot.com/configuration-options/#matchcurrentversion

**Reply by rarkins** [reply]
@viceice I think what he's after is a rule based on `currentVersion`, not `currentValue`.

@andreibuntsev we don't have any automatic way to say "upgrade only ranges and not single versions". If you have 200 packages where you've decided to never upgrade then I don't think that's a great idea. Ignoring particular dependencies or versions should ideally only be  workaround in a few cases, not hundreds.

**Reply by andreibuntsev** [reply]
Thank you guys, I'll think about that
